app/controllers/api/v1/hotels_controller.rb

```diff
@@ -4,22 +4,41 @@ class Api::V1::HotelsController < ApplicationController
 
   # GET /api/v1/hotels
   def index
-    @hotels = Hotel.all
+    # @hotels = Hotel.all.with_attached_images
+    @hotels = []
+    # render json: @hotels
+    Hotel.all.each do |hotel|
+      @hotels.push(hotel.as_json.merge({ image: url_for(hotel.image) }))
+    end
 
-    render json: @hotels
+    if @hotels.size.positive?
+      render json: @hotels
+    else
+      render json: { errors: 'Hotels not found' }, status: :not_found
+    end
   end
 
   # GET /api/v1/hotels/1
   def show
+    @result = []
     @hotel = Hotel.find(params[:id])
-    render json: @hotel
+    @result.push(@hotel.as_json.merge({ image: url_for(@hotel.image) }))
+
+    render json: @result
```

Comment:
@Tadessejemal1 Good job on implementing this functionality!!! 🔥 It works just fine. I learned something new today 🧑‍🎓 .